viam_cartographer.go

```diff
@@ -231,9 +232,12 @@ func New(
 	}
 
 	// Get the odometer if it is supported
+	var timedOdometer s.TimedOdometerSensor
 	if odometerSupported {
-		// TODO[kat]: Add when working on https://viam.atlassian.net/browse/RSDK-4860
-		logger.Info("odometer is not yet supported")
+		timedOdometer, err = s.NewOdometer(ctx, deps, movementSensorName, optionalConfigParams.MovementSensorDataFrequencyHz, logger)
```

Comment:
[opt] I'm a little concerned about making both an IMU and an odometer out of the same movement sensor. I think this means we are going to have two separate threads that read from the same movement sensor, which may block each other. I think it would be better to have a single thread that reads from the movement sensor and gets both odometry and IMU data.

Reply:
That's a good point. I thought about this and discussed it with @kim-mishra & @jeremyrhyde . We decided to use only one process for the movement sensor and then split it into IMU & odometer data within [`tryAddIMUReading`](https://github.com/viamrobotics/viam-cartographer/blob/3905758c2a20a80a09dd19bde97441f1fb92448a/sensorprocess/imusensorprocess.go#L132-L146) (which we'd rename to `tryAddMovementSensorReading`)

Reply:
Sounds good! Does it still make sense to have separate IMU and Odometer structs?

Reply:
No, I don't think so. It'll take a bit of refactoring to change this. Working on it now!